internal/view/diagnostic/function.go

```diff
@@ -0,0 +1,120 @@
+package diagnostic
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+// FunctionParam represents a single parameter to a function, as represented by type Function.
+type FunctionParam struct {
+	// Name is a name for the function which is used primarily for documentation purposes.
+	Name string `json:"name"`
+
+	// Type is a type constraint which is a static approximation of the possibly-dynamic type of the parameter
+	Type json.RawMessage `json:"type"`
+
+	Description     string `json:"description,omitempty"`
+	DescriptionKind string `json:"description_kind,omitempty"`
+}
+
+func DescribeFunctionParam(p *function.Parameter) FunctionParam {
+	ret := FunctionParam{
+		Name: p.Name,
+	}
+
+	if raw, err := p.Type.MarshalJSON(); err != nil {
+		// Treat any errors as if the function is dynamically typed because it would be weird to get here.
+		ret.Type = json.RawMessage(`"dynamic"`)
+	} else {
+		ret.Type = json.RawMessage(raw)
```

Comment:
Seems to be redundant type conversion  (same in other places too)

Reply:
Fixed